applications/ShapeOptimizationApplication/custom_utilities/optimization_utilities.h

```diff
@@ -318,6 +319,112 @@ class OptimizationUtilities
         return mCorrectionScaling;
     }
 
+    /**
+     * Assemble the values of the nodal vector variable into a vector
+     */
+    void AssembleVector(
+        Vector& rVector,
+        const Variable<array_3d> &rVariable)
+    {
+        if (rVector.size() != mrDesignSurface.NumberOfNodes()*3){
+            rVector.resize(mrDesignSurface.NumberOfNodes()*3);
+        }
+
+        int i=0;
+        for (auto & node_i : mrDesignSurface.Nodes())
+        {
+            array_3d& variable_vector = node_i.FastGetSolutionStepValue(rVariable);
+            rVector[i*3+0] = variable_vector[0];
+            rVector[i*3+1] = variable_vector[1];
+            rVector[i*3+2] = variable_vector[2];
+            ++i;
+        }
+    }
+
+    /**
+     * Assigns the values of a vector to the nodal vector variables
+     */
+    void AssignVectorToVariable(
```

Comment:
I don't know how often I have already implemented this exact functions :D

Time to move it to the core (not saying you should do it, just saying that many could benefit)